canplayer.c

```diff
@@ -252,6 +260,28 @@ static int add_assignment(char *mode, int socket, char *txname,
 	return 0;
 }
 
+void load_gaps_from_file(FILE* fp, sleep_struct *a) {
+    char *line = NULL;
+	size_t len = 0;
+	ssize_t read;
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		if (a->idx == a->size) {
+			a->size += 1;
+			a->sleep_vector = realloc(a->sleep_vector, a->size * sizeof(struct timespec));
```

Comment:
You're allocating `struct timespec`, but you need `struct timeval`.

```suggestion
			a->sleep_vector = realloc(a->sleep_vector, a->size * sizeof(*(s->sleep_vector)));
```